stdlib/REPL/src/docview.jl

```diff
@@ -299,8 +301,33 @@ function summarize(io::IO, T::DataType, binding::Binding)
     end
 end
 
-function summarize(io::IO, m::Module, binding::Binding)
-    println(io, "No docstring found for module `", m, "`.\n")
+function find_readme(m::Module)::Union{String, Nothing}
+    isnothing(pathof(m)) && return nothing
+    path = dirname(pathof(m))
+    top_path = pkgdir(m)
+    while true
+        for readme in ["README.md", "readme.md", "Readme.md", "ReadMe.md"]
+            readme_path = joinpath(path, readme)
+            if isfile(readme_path)
+                return readme_path
+            end
+        end
+        path == top_path && break # go no further than pkgdir
+        path = dirname(path) # work up through nested modules
+    end
+    return nothing
+end
+function summarize(io::IO, m::Module, binding::Binding; nlines::Int = 200)
+    readme_path = find_readme(m)
+    if !isnothing(readme_path)
+        println(io, "No docstring found for module `", m, "`.\n")
+        readme_lines = readlines(readme_path)
+        println(io, "Displaying the contents of `$(readme_path)`:\n\n")
+        [println(io, line) for line in first(readme_lines, nlines)]
```

Comment:
This seems a weird place to use a list comprehennsion.
I could understand using a broadcast because it is short, but a list comprehension just seems odd.

```suggestion
        println.(io, first(readme_lines, nlines))
```
or
```suggestion
        println(io, join(first(readme_lines, nlines), "\n")))
```
or probably best:
```suggestion
        for line in line in first(readme_lines, nlines)
            println(io, line)
        end
```